server/chat/backend/agent/tools/bitbucket/utils.py

```
@@ -0,0 +1,175 @@
+
+            old_access_token = access_token
+            bb_creds = refresh_token_if_needed(bb_creds)
+            access_token = bb_creds.get("access_token", access_token)
+
+            # Persist refreshed token if changed
+            if access_token != old_access_token:
+                try:
+                    from utils.auth.token_management import store_tokens_in_db
+                    store_tokens_in_db(user_id, bb_creds, "bitbucket")
+                    logger.info(f"Persisted refreshed Bitbucket token for user {user_id}")
+                except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)